org.lflang.tests/src/org/lflang/tests/cli/LfcCliTest.java

```diff
@@ -101,10 +106,56 @@ public void testGenInSrcDir(@TempDir Path tempDir) throws IOException {
 
     }
 
+    @Test
+    public void testTargetProperties(@TempDir Path tempDir)
+            throws IOException {
+        dirBuilder(tempDir).file("src/File.lf", LF_PYTHON_FILE);
 
-    static class LfcTestFixture extends CliToolTestFixture {
+        String[] args = {
+            "src/File.lf",
+            "--output-path", "src",
+            "--build-type", "Release",
+            "--clean",
+            "--target-compiler", "gcc",
+            "--external-runtime-path", "src",
+            "--federated",
+            "--logging", "4",
+            "--lint",
+            "--no-compile",
+            "--quiet",
+            "--rti", "-1",
+            "--runtime-version", "rs",
+            "--scheduler", "2",
+            "--threading", "false",
+            "--workers", "1",
+        };
+
+        lfcTester.runLfcObj(tempDir, args)
+            .verify(result -> {
+                result.checkOk();
+                // Get the properties method.
+                Method getPropsMethod =
```

Comment:
Calling methods by reflection in test code is an anti pattern. It makes usage search and static type checking useless. Even though the method is an implementation detail, I think the lesser evil is definitely to make it public.